### PR TITLE
[1LP][RFR] SCVMM test cleanup

### PR DESCRIPTION
**Pull request by john-dupuy**
The purpose of this pr is to:

1) Add testSteps for manual cases in `test_scvmm_specific`, these will be automated later
2) Remove tests that have 2016 in the name, these tests will be parametrized against both SCVMM 2016 and 2012
3) Minor fixes for `test_no_dvd_ruins_refresh`

{{ pytest: --long-running --use-template-cache --use-provider complete cfme/tests/infrastructure/test_scvmm_specific.py }}

### CODE REVIEW

File: cfme/tests/infrastructure/test_scvmm_specific.py [hunk 1]
```diff
@@ -81,6 +76,12 @@ def test_vm_mac_scvmm():
         caseimportance: low
         initialEstimate: 1/20h
         setup: https://bugzilla.redhat.com/show_bug.cgi?id=1514461
+        testSteps:
+            1. Add SCVMM to CFME
```
Comment (ganeshhubale) [inline cfme/tests/infrastructure/test_scvmm_specific.py]: It would be better to write description or steps for ```setup``` rather than providing BZ link. Although you have already mentioned BZ above.

File: cfme/tests/infrastructure/test_scvmm_specific.py [hunk 2]
```diff
@@ -130,17 +115,23 @@ def test_provider_summary_scvmm():
     """
     The purpose of this test is to verify that the information on the
     provider summary is substantially the same as what is on SCVMM.
-    Since SCVMM-SP1 only has a short sequence of test cases, you must use
-    this test case as the catch all to go in and spend 15-30 minutes and
-    check as many links from this page and verify both the navigation and
-    the content.
+    Note: when automated this test will likely be broken up into many tests.
+
```
Comment (ganeshhubale) [inline cfme/tests/infrastructure/test_scvmm_specific.py]: Remove extra blank line no. 119

File: cfme/tests/infrastructure/test_scvmm_specific.py [hunk 3]
```diff
@@ -130,17 +115,23 @@ def test_provider_summary_scvmm():
...
 
     Polarion:
         assignee: jdupuy
         casecomponent: Infra
         caseimportance: medium
         initialEstimate: 1/2h
-        startsin: 5.4
+        startsin: 5.7
+        testSteps:
```
Comment (ganeshhubale) [inline cfme/tests/infrastructure/test_scvmm_specific.py]: Question(Just curious):
- Why did you change ```startsin``` version?

File: cfme/tests/infrastructure/test_scvmm_specific.py [hunk 4]
```diff
@@ -149,34 +140,20 @@ def test_provider_summary_scvmm():
 @pytest.mark.tier(2)
 def test_host_info_scvmm():
     """
-    The purpose of this test is to verify that SCVMM-SP1 hosts are not
-    only added, but that the host information details are correct.  Take
-    the time to spot check at least one host.
+    Testing that host info is correct in CFME
```
Comment (ganeshhubale) [inline cfme/tests/infrastructure/test_scvmm_specific.py]: I can see mistake in this sentence here, as I went through your steps which are very clear; This sentence should be like
``` Testing the host info is correct in CFME```(if I am not wrong - current sentence indicates that you are talking about specific host)

File: cfme/tests/infrastructure/test_scvmm_specific.py [hunk 5]
```diff
@@ -186,37 +163,27 @@ def test_host_info_scvmm2016():
...
-    """
-    pass
-
-
-@pytest.mark.manual
-@pytest.mark.tier(2)
-def test_vm_volume_specchar1_scvmm():
-    """
-    Special Test to verify that VMs that have Volumes with no drive letter
-    assigned don"t cause systemic SCVMM provider errors.  This is a low
```
Comment (ganeshhubale) [inline cfme/tests/infrastructure/test_scvmm_specific.py]: Question:
Is this test case automated or covers in other test case?

File: cfme/tests/infrastructure/test_scvmm_specific.py [hunk 6]
```diff
@@ -149,34 +140,20 @@ def test_provider_summary_scvmm():
 @pytest.mark.tier(2)
 def test_host_info_scvmm():
     """
-    The purpose of this test is to verify that SCVMM-SP1 hosts are not
-    only added, but that the host information details are correct.  Take
```
Comment (ganeshhubale) [inline cfme/tests/infrastructure/test_scvmm_specific.py]: Add ```test_requirements``` to this test case and also see other test cases in this file.

File: cfme/tests/infrastructure/test_scvmm_specific.py [hunk 7]
```diff
@@ -6,6 +6,12 @@
 from cfme.infrastructure.provider.scvmm import SCVMMProvider
 
 
+pytestmark = [
+    pytest.mark.provider([SCVMMProvider], scope="module"),
+    pytest.mark.usefixtures("setup_provider_modscope")
+]
+
```
Comment (ganeshhubale) [inline cfme/tests/infrastructure/test_scvmm_specific.py]: Question:
- We have already provided ```scope="module"``` for ```SCVMMProvider``` then why do we need ```setup_provider_modscope```? Cant we use ```setup_provider```?

File: cfme/tests/infrastructure/test_scvmm_specific.py [hunk 8]
```diff
@@ -46,24 +50,13 @@ def test_template_info_scvmm2016():
         caseimportance: medium
         initialEstimate: 1/8h
         startsin: 5.7
-    """
-    pass
 
-
-@pytest.mark.manual
```
Comment (ganeshhubale) [inline cfme/tests/infrastructure/test_scvmm_specific.py]: Remove this blank line no.53

File: cfme/tests/infrastructure/test_scvmm_specific.py (hunk 1 above)
Comment (john-dupuy) [reply cfme/tests/infrastructure/test_scvmm_specific.py]: Actually this `setup` is unnecessary, as it's covered by the `testSteps`.

File: cfme/tests/infrastructure/test_scvmm_specific.py (hunk 7 above)
Comment (john-dupuy) [reply cfme/tests/infrastructure/test_scvmm_specific.py]: As far as I know, I think it is still necessary to use `setup_provider_modscope`. https://github.com/ManageIQ/integration_tests/blob/d6be60884001d80ddf5e5320beb5ad7f5cf7d0fc/cfme/fixtures/provider.py#L284

File: cfme/tests/infrastructure/test_scvmm_specific.py (hunk 3 above)
Comment (john-dupuy) [reply cfme/tests/infrastructure/test_scvmm_specific.py]: Hmm... actually not sure about that one, probably to match the other startsin? I will leave it as 5.4

File: cfme/tests/infrastructure/test_scvmm_specific.py (hunk 4 above)
Comment (john-dupuy) [reply cfme/tests/infrastructure/test_scvmm_specific.py]: Good catch, I forgot _the_ article! I'll make it read, `Testing to make sure that the host info is correct in CFME`

File: cfme/tests/infrastructure/test_scvmm_specific.py (hunk 5 above)
Comment (john-dupuy) [reply cfme/tests/infrastructure/test_scvmm_specific.py]: Nope, not yet automated, I plan to automate most of these in the near future. This could be considered to be covered by `test_provider_summary_scvmm`, but since this test case is specific to a BZ, I think it should be its own test case.

File: cfme/tests/infrastructure/test_scvmm_specific.py [hunk 9]
```diff
@@ -34,7 +38,8 @@ def test_no_dvd_ruins_refresh(provider, testing_vm_without_dvd):
 
 @pytest.mark.manual
 @pytest.mark.tier(2)
-def test_template_info_scvmm2016():
+@test_requirements.general_ui
```
Comment (lina-nikiforova) [inline cfme/tests/infrastructure/test_scvmm_specific.py]: Q: are you sure it should belong to general UI?

Comment (john-dupuy) [reply cfme/tests/infrastructure/test_scvmm_specific.py]: I'm not sure what else it would be, since this test is to make sure that all the information is displayed properly in the UI. Is there a `test_requirement` you think is more appropriate?

**Comment by lina-nikiforova** [comment]
LGTM, except that no tests ran in PRT. Why is that?

File: cfme/tests/infrastructure/test_scvmm_specific.py (hunk 9 above)
Comment (lina-nikiforova) [reply cfme/tests/infrastructure/test_scvmm_specific.py]: I would add provider-specific requirements to #8672 
What do you think?

Comment (john-dupuy) [reply cfme/tests/infrastructure/test_scvmm_specific.py]: Yep, I pinged Mike and he added them, so this PR is waiting on #8672 to be merged.

Comment (mshriver) [reply cfme/tests/infrastructure/test_scvmm_specific.py]: So there are a few things going on here, and what I'm going to suggest doesn't have to happen in this PR specifically.

We shouldn't have such small test cases, written for specific providers, for something so generic.

We already have a ton of automation around this exact thing - its called 'Relationship Discovery', and you'll find test modules called 'test_relationships'.  These should be generically automated to check all relationship fields in the UI for a given provider, comparing the counts against wrapanapi's queries to the provider directly.

There are already framework components in place for all of this, implemented under the 'validate_stats' method stacks.

If anything this case should be altogether removed, and the SCVMM provider included in test collection/parametrization for the existing automated relationship tests..

Comment (mshriver) [reply cfme/tests/infrastructure/test_scvmm_specific.py]: Ultimately, as far as test metadata goes, I would probably consider this type of test a 'relationships' marker (new in 8672, but old FA that never had requirement marker).  I'd probably give it a casecomponent of 'WebUI' since its also just validating the webUI is displaying things correctly.

Comment (john-dupuy) [reply cfme/tests/infrastructure/test_scvmm_specific.py]: @mshriver definitely agreed, this PR is more focused on cleaning up and adding metadata to the existing SCVMM tests. 

In the next sprint, I plan to go through each of these test cases and evaluate (1) whether or not it is necessary and (2) whether or not it can be automated. If it is able to be automated then I write a test for it. 

With that said, I'd like to leave these in here for now so that I know what to test for sign-off purposes.

**Comment by john-dupuy** [comment]
Failing travis right now due to #8672

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit cf1ab714647e953928817960c7f1dbea78b594b9:

Lint OK! :cake: :punch: :cookie: .

* cf1ab714647e953928817960c7f1dbea78b594b9
  * Merge commit detected!!! :bangbang:

**The PR might not be acceptable in this state** :rage4:.
There is a merge commit in the branch which we do not like in PRs so please, get rid of it.

*CFME QE Bot*

File: cfme/tests/infrastructure/test_scvmm_specific.py [hunk 10]
```diff
@@ -34,7 +39,7 @@ def test_no_dvd_ruins_refresh(provider, testing_vm_without_dvd):
 
 @pytest.mark.manual
 @pytest.mark.tier(2)
-def test_template_info_scvmm2016():
+def test_template_info_scvmm():
```
Comment (digitronik) [inline cfme/tests/infrastructure/test_scvmm_specific.py]: optional:
If you are trying to give good `id` then john best hack is inherit `provider` :)
